From 382e6c5e6ddbff7879f2c022f1ed71d48cc2541e Mon Sep 17 00:00:00 2001
From: Ronald Claveau <linux-kernel-dev@aliel.fr>
Date: Wed, 22 Apr 2026 11:43:30 +0200
Subject: [PATCH 90/91] arm64: dts: amlogic: t7: Add PCIE controller node

Describe the T7 PCIe-A host controller. The instance is based on the
DesignWare IP, reuses the existing Meson PCIe driver.

Constrain inbound DMA to the bottom 3.5 GiB to reflect the hardware
limitation enforced by BL31.

Signed-off-by: Ronald Claveau <linux-kernel-dev@aliel.fr>
Signed-off-by: Nick Xie <nick@khadas.com>
Change-Id: I2cf2a0b94fa874deaafccbff159ad0cd600fa752
---
 arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi | 34 +++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi b/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi
index 0a724d0ce5b7..2fffdecc81ac 100644
--- a/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi
+++ b/arch/arm64/boot/dts/amlogic/amlogic-t7.dtsi
@@ -428,6 +428,40 @@ gic: interrupt-controller@fff01000 {
 			interrupts = <GIC_PPI 9 (GIC_CPU_MASK_SIMPLE(8) | IRQ_TYPE_LEVEL_HIGH)>;
 		};
 
+		pcieA: pcie@f5000000 {
+			compatible = "amlogic,t7-pcie", "snps,dw-pcie";
+			reg = <0x0 0xf5000000 0x0 0x400000>,
+			      <0x0 0xfe02c000 0x0 0x2000>,
+			      <0x0 0xf5400000 0x0 0x200000>;
+			reg-names = "dbi", "cfg", "config";
+			interrupts = <GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "msi";
+			#interrupt-cells = <1>;
+			interrupt-map-mask = <0 0 0 0>;
+			interrupt-map = <0 0 0 0 &gic GIC_SPI 143 IRQ_TYPE_EDGE_RISING>;
+			bus-range = <0x0 0xff>;
+			#address-cells = <3>;
+			#size-cells = <2>;
+			device_type = "pci";
+			ranges = <0x81000000 0x0 0x0 0x0 0xf5600000 0x0 0x100000>,
+				 <0x82000000 0x0 0xf5700000 0x0 0xf5700000 0x0 0x1900000>;
+
+			clocks = <&clkc_periphs CLKID_SYS_PCIE_PHY>, <&clkc_periphs CLKID_SYS_PCIE>,
+				 <&pcie CLKID_PCIE_PLL>;
+			clock-names = "general", "pclk", "port";
+			power-domains = <&pwrc PWRC_T7_PCIE_ID>;
+			resets = <&reset RESET_PCIE_A>,
+				 <&reset RESET_PCIE_APB>;
+			reset-names = "port",
+				      "apb";
+
+			dma-ranges = <0x02000000 0x0 0x0 0x0 0X0 0x0 0xe0000000>;
+			num-lanes = <1>;
+			phys = <&pcie_phy PHY_TYPE_PCIE>;
+			phy-names = "pcie";
+			status = "disabled";
+		};
+
 		apb4: bus@fe000000 {
 			compatible = "simple-bus";
 			reg = <0x0 0xfe000000 0x0 0x480000>;
-- 
2.53.0

